File: Sort an array of 0s, 1s and 2s - GFG/sort-an-array-of-0s-1s-and-2s.c
// { Driver Code Starts
//Initial Template for C

#include <stdio.h>


 // } Driver Code Ends
//User function Template for C

void sort012(int a[], int n)
{
    int z=0,o=0,t=0;
    for(int i =0;i<n;i++)
    {
        if(a[i]==0) z++;
        else if(a[i]==1) o++;
        else if(a[i]==2) t++;
        
    }
    for(int i=0;i<n;i++)
    {
        if(z>0)
        {
            a[i]=0;
            z--;
        }
        else if(o>0)
        {
            a[i]=1;
            o--;
        }
        else if(t>0)
        {
            a[i]=2;
            t--;
        }
    }
    
}

// { Driver Code Starts.

int main() {

    int t;
    scanf("%d", &t);

    while(t--){
        int n;
        scanf("%d", &n);
        int arr[n];
        for(int i=0;i<n;i++){
            scanf("%d", &arr[i]);
        }

        sort012(arr, n);

        for (int i = 0; i < n; i++)
            printf("%d ", arr[i]);
        printf("\n");
    }
    return 0;
}
  // } Driver Code Ends

File: Sort an array of 0s, 1s and 2s - GFG/sort-an-array-of-0s-1s-and-2s.java
//{ Driver Code Starts
//Initial template for Java

import java.io.*;
import java.util.*;


// } Driver Code Ends
//User function template for Java

class Solution
{
    public static void sort012(int arr[], int n)
    {
        // code here 
        int zero = 0,one = 0, two = 0;
        for(int i :arr)
        {
            if(i==0) zero++;
            if(i==1) one++;
            if(i==2) two++;
        }
        for(int i = 0 ;i<n;i++)
        {
            if(zero>0)
            {
                arr[i]=0;
                zero--;
            }
            else if(one>0)
            {
                arr[i]=1;
                one--;
            }
            else if(two>0)
            {
                arr[i] = 2;
                two--;
            }
        }
    }
}

//{ Driver Code Starts.

class GFG {
    
    public static void main (String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim()); //Inputting the testcases
        while(t-->0){
            int n = Integer.parseInt(br.readLine().trim());
            int arr[] = new int[n];
            String inputLine[] = br.readLine().trim().split(" ");
            for(int i=0; i<n; i++){
                arr[i] = Integer.parseInt(inputLine[i]);
            }
            Solution ob=new Solution();
            ob.sort012(arr, n);
            StringBuffer str = new StringBuffer();
            for(int i=0; i<n; i++){
                str.append(arr[i]+" ");
            }
            System.out.println(str);
        }
    }
}


// } Driver Code Ends